src/client/components/github.ts

```diff
@@ -0,0 +1,434 @@
+import { LitElement, css, html } from 'lit'
+import { customElement, property } from 'lit/decorators.js'
+import { when } from 'lit/directives/when.js'
+import { Disposable } from 'vscode'
+
+import type { ClientMessage, GitHubState } from '../../types'
+import { getContext } from '../utils'
+import { ClientMessages } from '../../constants'
+import { onClientMessage } from '../../utils/messaging'
+import { IWorkflowRun } from '../../extension/services/types'
+
+
+interface IWorkflow {
+  on: any
+  name: string
+  jobs: any
+}
+
+type Event = {
+  target: {
+    id: string
+    value: string
+  }
+}
+
+enum DeploymentStatus {
+  triggered,
+  error,
+  none
+}
+
+@customElement('github-output')
+export class GitHubOutput extends LitElement{
+
+  @property()
+  protected isTriggeringWorkflow = false
+
+  @property()
+  protected deploymentStatus: DeploymentStatus = DeploymentStatus.none
+
+  @property()
+  protected reason?: string
+
+  @property()
+  workflowRun?: IWorkflowRun
+
+  @property({ type: Object })
+  state: GitHubState = {}
+  
+  protected disposables: Disposable[] = []
+
+  private inputs: Record<string, string> = {}
+
+  // Define scoped styles right with your component, in plain CSS
+  /* eslint-disable max-len */
+  static styles = css`
```

Comment:
I'd suggest moving this to a `styles.ts` or similar file ahead of a full review @degrammer.

Reply:
Moved to a css file and added a webpack raw-loader, it provides better experience for editing CSS